packages/main/src/plugin/container-registry.ts

```diff
@@ -621,6 +622,13 @@ export class ContainerProviderRegistry {
           return fetchedImages;
         }
 
+        // Using guessIsManifest, determine if the image is a manifest and set isManifest accordingly
+        // NOTE: This is a workaround until we have a better way to determine if an image is a manifest
+        // and may result in false positives until issue: https://github.com/containers/podman/issues/22184 is resolved
+        fetchedImages.forEach(image => {
+          image.isManifest = guessIsManifest(image);
+        });
+
```

Comment:
maybe get rid of this forEach and move the lines we have after this block as we iterate again on all the images

add only isManifest field by doing something like

```ts
        // Transform fetched images to include engine name and ID
        return fetchedImages.map(image => ({
          ...image,
          engineName: provider.name,
          engineId: provider.id,
        // Using guessIsManifest, determine if the image is a manifest and set isManifest accordingly
        // NOTE: This is a workaround until we have a better way to determine if an image is a manifest
        // and may result in false positives until issue: https://github.com/containers/podman/issues/22184 is resolved
         isManifest: guessIsManifest(image);
        }));

```

Reply:
You are completely right, it should be in here, 🤦